src/Tools/dotnet-monitor/CollectionRules/CollectionRuleContext.cs

```diff
@@ -31,5 +33,12 @@ public CollectionRuleContext(string name, CollectionRuleOptions options, IEndpoi
         public CollectionRuleOptions Options { get; }
 
         public string Name { get; }
+
+        /// <summary>
+        /// Note that we only reference named actions. As such, unnamed actions will not save their results for other actions
+        /// to consume.
+        /// </summary>
+        public IDictionary<string, CollectionRuleActionResult> ActionResults { get; } =
```

Comment:
I'm not a fan of putting this on the CollectionRuleContext. These values only represent the output of the actions of a specific action list invocation, not for the entire lifetime of the rule. Additionally, everything on this context is largely just for reference and not for mutation.

If you keep this here, you absolutely need to fix the fact that these results are not reset between runs of the action list inside of the CollectionRulePipeline. Again, I don't like the idea of the pipeline having to need to know how to set up the state of the action list execution.

Reply:
Alternatively, return this dictionary from the ExecuteActions method rather than stashing on the context.